raft/server.cc

```diff
@@ -444,6 +468,12 @@ future<> server_impl::wait_for_entry(entry_id eid, wait_type type, seastar::abor
                 // and we don't know if the entry was applied with `state_machine::apply`
                 // (we may've loaded a snapshot before we managed to apply the entry).
                 // As specified by `add_entry`, throw `commit_status_unknown` in this case.
+                //
+                // FIXME: replace this with a different exception type - `commit_status_unknown`
+                // gives too much uncertainty while we know that the entry was committed
+                // and had to be applied on at least one server. Some callers of `add_entry`
+                // need to know only that the current state includes that entry, whether it was done
+                // through `apply` on this server or through receiving a snapshot.
```

Comment:
We may add a flag to commit_status_unknown may be.

Reply:
Or we could change the future we're waiting on here to the future that is waiting for the snapshot apply.